sdk/identity/src/client_credentials_flow/mod.rs

```diff
@@ -39,13 +39,18 @@
 
 mod login_response;
 
-use azure_core::error::{ErrorKind, Result, ResultExt};
+use azure_core::{
+    content_type,
+    error::{ErrorKind, Result, ResultExt},
+    headers, HttpClient, Request,
+};
+use http::Method;
 use login_response::LoginResponse;
 use url::form_urlencoded;
 
 /// Perform the client credentials flow
 pub async fn perform(
-    client: reqwest::Client,
+    http_client: &dyn HttpClient,
```

Comment:
This feels weird from the consumer side, given the client we get from `new_http_client` is a `Arc<dyn HttpClient>`.   In other areas in the API, we use `.clone()` instead of `.as_ref()`, which feels slightly more natural.

Reply:
Other than that, this LGTM and works in all of my use cases.